<?xml version="1.0" ?>
<robot name="jdamr">
    <!-- robot body link  -->
    <link name="body_link">
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0"/> <!-- rpy: roll pitch yaw origin: origin point-->
            <geometry>
                <box size="1 0.6 0.3"/>  <!-- box size -->
            </geometry>
            <material name="red">
                <color rgba="1 0 0 1"/>
            </material>
        </visual>
    </link>
    <!-- wheel 1 joint   -->
     <joint name="wheel1_joint" type="continuous">
        <parent link="body_link"/>
        <child link="wheel1_link" />
        <origin xyz="-0.3 -0.35 -0.1" rpy="0 0 0"/>
        <axis xyz="0 1 0" />
    </joint>
     <!-- wheel 1 link   -->
    <link name="wheel1_link">
        <visual>
            <origin rpy="1.570795 0 0" xyz="0 0 0"/>
            <geometry>
                <cylinder length="0.1" radius="0.15"/>
            </geometry>
            <material name="yellow">
                <color rgba="1 1 0 1"/>
            </material>
        </visual>
    </link>
    <!-- wheel 2 joint   -->
    <joint name="wheel2_joint" type="continuous">
        <parent link="body_link"/>
        <child link="wheel2_link" />
        <origin xyz="0.3 -0.35 -0.1" rpy="0 0 0"/>
        <axis xyz="0 1 0" />
    </joint>
    <!-- wheel 2 link   -->
    <link name="wheel2_link">
        <visual>
            <origin rpy="1.570795 0 0" xyz="0 0 0"/>
            <geometry>
                <cylinder length="0.1" radius="0.15"/>
            </geometry>
            <material name="yellow">
                <color rgba="1 1 0 1"/>
            </material>
        </visual>
    </link>
    <!-- wheel 3 joint   -->
    <joint name="wheel3_joint" type="continuous">
        <parent link="body_link"/>
        <child link="wheel3_link" />
        <origin xyz="0.3 0.35 -0.1" rpy="0 0 0"/>
        <axis xyz="0 1 0" />
    </joint>
    <!-- wheel 3 link   -->
    <link name="wheel3_link">
        <visual>
            <origin rpy="1.570795 0 0" xyz="0 0 0"/>
            <geometry>
                <cylinder length="0.1" radius="0.15"/>
            </geometry>
            <material name="yellow">
                <color rgba="1 1 0 1"/>
            </material>
        </visual>
    </link>
     <!-- wheel 4 joint   -->
    <joint name="wheel4_joint" type="continuous">
        <parent link="body_link"/>
        <child link="wheel4_link" />
        <origin xyz="-0.3 0.35 -0.1" rpy="0 0 0"/>
        <axis xyz="0 1 0" />
    </joint>
    <!-- wheel 4 link   -->
    <link name="wheel4_link">
        <visual>
            <origin rpy="1.570795 0 0" xyz="0 0 0"/>
            <geometry>
                <cylinder length="0.1" radius="0.15"/>
            </geometry>
            <material name="yellow">
                <color rgba="1 1 0 1"/>
            </material>
        </visual>
    </link>
</robot>